remoteworkitem/flatten_test.go

```diff
@@ -0,0 +1,25 @@
+package remoteworkitem
+
+import (
+	"encoding/json"
+	"testing"
+
+	"github.com/almighty/almighty-core/resource"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestFlattenMap(t *testing.T) {
+	resource.Require(t, resource.UnitTest)
+	testString := []byte(`{"name":"shoubhik","foo":[1,2,3,4], "assignee":{"1":"sbose","2":"pranav","participants":{"4":"sbose56","5":"sbose78"}},"name":"shoubhik"}`)
+	var nestedMap map[string]interface{}
+	err := json.Unmarshal(testString, &nestedMap)
+
+	if err != nil {
+		t.Error("Incorrect dataset ", testString)
+	}
+	fm := MapFlattener{}
+	OneLevelMap := fm.Flatten(nestedMap)
+	if !assert.Equal(t, OneLevelMap["assignee.participants.4"], "sbose56") {
```

Comment:
> Add some tests to verify that 'flatten' handles the correct data types as well, e.g. string, int, bool, array

Thanks, 
Added assert statments. Just slightly concerned about how the unmarshalling converts int to float64. Let me know if that needs to be handled in a better way.